src/app/components/form/form.component.ts

```diff
@@ -51,21 +52,38 @@ export class FormComponent implements OnInit, OnChanges, OnDestroy {
   private _showProgressBar: boolean;
   private langChangeSubscription: Subscription;
 
+  private domainName: string = '';
+  private routeParamsSubscription: Subscription;
+
   public history = {};
   public test = {};
   public disable_check_button = false;
   public form: FormGroup;
 
-  constructor(private formBuilder: FormBuilder, private translateService: TranslateService, private titleService: Title) {}
+  constructor(private activatedRoute: ActivatedRoute,
+    private formBuilder: FormBuilder,
+    private translateService: TranslateService,
+    private titleService: Title) {}
 
   ngOnInit() {
+    this.routeParamsSubscription = this.activatedRoute.params.subscribe((params: Params) => {
+      if ( params['domain'] ) {
+        this.domainName = params['domain'];
+      }
+    });
+
     this.langChangeSubscription = this.translateService.onLangChange.subscribe((_: LangChangeEvent) => {
       if (this.form.touched) {
         this.runDomainCheck(false);
       }
     });
+
     this.generate_form();
     this.titleService.setTitle('Zonemaster');
+
+    if ( this.domainName ) {
+        this.runDomainCheck();
```

Comment:
The `runDomainCheck` should be call from the subscribe callback, otherwise it will not be triggered if route parameter is changed (e.g. changing the domain in url and going back in history to the first one).

Reply:
fixed